__tests__/app.test.js

```diff
@@ -32,142 +32,263 @@ describe("GET/api/topics", () => {
-              body: expect.any(String),
-              created_at: expect.any(String),
-              votes: expect.any(Number),
-              comment_count: expect.any(Number),
-            })
-          );
-        })   
+        expect(article).toHaveLength(12);
+        expect(article).toBeSortedBy("created_at", {
+          descending: true,
+        });
       });
   });
 });
-
-describe("GET/api/articles?topic=mitch", () => {
+describe("GET/api/articles?topic", () => {
   test("200: return all the articles with topic mitch sort by date descending", () => {
     return request(app)
       .get("/api/articles?topic=mitch")
       .expect(200)
       .then(({ body: article }) => {
-        expect(article).toHaveLength(23);
-        article.forEach((article) => {
-          expect(article).toEqual(
-            expect.objectContaining({
-              article_id: expect.any(Number),
-              title: expect.any(String),
-              topic: expect.any(String),
-              author: expect.any(String),
-              body: expect.any(String),
-              created_at: expect.any(String),
-              votes: expect.any(Number),
-              comment_count: expect.any(Number),
-            })
-          );
+        expect(article).toHaveLength(11);
+        expect(article).toBeSortedBy("created_at", {
+          descending: true
         });
       });
   });
+  test("404: return error for invalid query", () => {
+    return request(app)
+      .get("/api/articles?topic=banana")
+      .expect(404)
+      .then(({ body: article }) => {
+        expect(article.msg).toBe("Query not valid");
+      });
+  });
 });
 
-describe("GET/api/articles?topic=banana", () => {
-  test("200: return all the articles with topic mitch sort by date descending", () => {
+describe("GET/api/articles?sort_by", () => {
+  test("200: return all the articles with authors in descending order", () => {
     return request(app)
-      .get("/api/articles?topic=banana")
+      .get("/api/articles?sort_by=author")
+      .expect(200)
+      .then(({ body: article }) => {
+        expect(article).toHaveLength(12);
+        expect(article).toBeSortedBy("author", {
+          descending: true,
+        });
+      });
+  });
+  test("200: return all the articles with authors in descending order", () => {
+    return request(app)
+      .get("/api/articles?sort_by=author&order=asc")
+      .expect(200)
+      .then(({ body: article }) => {
+        expect(article).toHaveLength(12);
+        expect(article).toBeSortedBy("author", {
+          descending: false,
+        });
+      });
+  });
+  test("200: return all the articles with dates in ascending order", () => {
+    return request(app)
+      .get("/api/articles?order=asc")
+      .expect(200)
+      .then(({ body: article }) => {
+        expect(article).toHaveLength(12);
+        expect(article).toBeSortedBy("created_at", {
+          descending: false,
+        });
+      });
+  });
```

Comment:
A 200 test that combines the topic filter with sort_by and order would be worthwhile here